lib/galaxy/tools/actions/upload_common.py

```diff
@@ -120,11 +120,14 @@ def persist_uploads(params, trans):
                                                    local_filename=local_filename)
             elif type(f) == dict and 'local_filename' not in f:
                 raise Exception('Uploaded file was encoded in a way not understood by Galaxy.')
-            if 'url_paste' in upload_dataset and upload_dataset['url_paste'] and upload_dataset['url_paste'].strip() != '':
-                upload_dataset['url_paste'] = datatypes.sniff.stream_to_file(
-                    StringIO(validate_url(upload_dataset['url_paste'], trans.app.config.fetch_url_whitelist_ips)),
-                    prefix="strio_url_paste_"
-                )
+
+            if 'url_paste' in upload_dataset and upload_dataset['url_paste']:
```

Comment:
```suggestion
            if 'url_paste' in upload_dataset:
```

Reply:
I think we need the second condition: 'url_paste' in dict tests the presence of the key; the value still may be None - in which case calling strip() in the value will raise an attributeerror, no?

Reply:
True, could be the case of None.